### DISCUSSION TITLE
Building rustdesk lib file 'librustdesk.so' failed, someone knows that, every step is currectly!

### DISCUSSION THREAD

**Discussion by LJRFox** [rustdesk/rustdesk#4208] [Q&A]
`
ddd@ubuntu:~/rustdesk/rustdesk$ ./flutter/ndk_arm64.sh
[2023-04-27T03:04:50Z INFO  cargo_ndk::cli] Using NDK at path: /home/ddd/Android/Sdk/ndk/23.2.8568313 (ANDROID_NDK_HOME)
[2023-04-27T03:04:50Z INFO  cargo_ndk::cli] NDK API level: 21
[2023-04-27T03:04:50Z INFO  cargo_ndk::cli] Building targets: arm64-v8a
[2023-04-27T03:04:50Z INFO  cargo_ndk::cli] Building arm64-v8a (aarch64-linux-android)
   Compiling rustdesk v1.2.0 (/home/ddd/rustdesk/rustdesk)
   Compiling magnum-opus v0.4.0 (https://github.com/rustdesk/magnum-opus#79be072c)
error: failed to run custom build command for `magnum-opus v0.4.0 (https://github.com/rustdesk/magnum-opus#79be072c)`

Caused by:
  process didn't exit successfully: `/home/ddd/rustdesk/rustdesk/target/release/build/magnum-opus-47936c2d16da1546/build-script-build` (exit status: 101)
  --- stdout
  cargo:info=arm64-android
  cargo:rustc-link-lib=static=opus
  cargo:rustc-link-search=/home/ddd/vcpkg/installed/arm64-android/lib
  cargo:include=/home/ddd/vcpkg/installed/arm64-android/include
  rerun-if-changed=/home/ddd/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/79be072/opus_ffi.h
  rerun-if-changed=/home/ddd/vcpkg/installed/arm64-android/include

  --- stderr
  /usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found
  /usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found, err: true
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', /home/ddd/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/79be072/build.rs:128:18
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
warning: function `gen_flutter_rust_bridge` is never used
  --> build.rs:80:4
   |
80 | fn gen_flutter_rust_bridge() {
   |    ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: `rustdesk` (build script) generated 1 warning
[2023-04-27T03:04:54Z INFO  cargo_ndk::cli] If the build failed due to a missing target, you can run this command:
[2023-04-27T03:04:54Z INFO  cargo_ndk::cli] 
[2023-04-27T03:04:54Z INFO  cargo_ndk::cli]     rustup target install aarch64-linux-android
`

**Comment by LJRFox** [comment]
The docs said the vcpkg directory to be at $HOME, I have set it in the right directory, BUT why does this problem still occur.

**Comment by deep-soft** [comment]
from:
[(https://github.com/rustdesk/rustdesk/tree/master/.github/workflows)/flutter-build.yml](https://github.com/rustdesk/rustdesk/tree/master/.github/workflows/flutter-build.yml)
you need: 
```

...
  build-rustdesk-android:
...

      - name: Clone deps
        shell: bash
        run: |
          pushd /opt
          git clone https://github.com/Kingtous/rustdesk_thirdparty_lib.git --depth=1

# and set 
          VCPKG_ROOT: /opt/rustdesk_thirdparty_lib/vcpkg
```

but is easy to use github CI to do the build for you.

If you want to build locally, you have to follow the steps already included in .yaml file, and adapt them to you're needs.

**Reply by LJRFox** [reply]
Thanks, extremely grateful, it's work for me!
